test/Validate.Capability.cpp

```diff
@@ -40,18 +40,77 @@
 namespace {
 
 using spvtest::ScopedContext;
+using std::get;
 using std::make_pair;
+using std::ostringstream;
 using std::pair;
 using std::string;
-using std::stringstream;
 using std::tuple;
 using std::vector;
 using testing::Combine;
 using testing::Values;
 using testing::ValuesIn;
 
-using ValidateCapability =
-    spvtest::ValidateBase<tuple<string, pair<string, vector<string>>>>;
+// Parameter for validation test fixtures.  Captures assembly to test in two
+// parts: a variable top line and a fixed remainder.  The top line will be an
+// OpCapability instruction, while the remainder will be some assembly text that
+// succeeds or fails to assemble depending on which capability was chosen.  For
+// instance, the following will succeed:
+//
+// OpCapability Pipes ; implies Kernel
+// OpLifetimeStop %1 0 ; requires Kernel
+//
+// and the following will fail:
+//
+// OpCapability Kernel
+// %1 = OpTypeNamedBarrier ; requires NamedBarrier
+//
+// So how does the test parameter capture which capabilities should cause
+// success and which shouldn't?  The answer is in the last element: it's a
+// vector of capabilities that make the remainder assembly succeed.  So if the
+// first-line capability exists in that vector, success is expected; otherwise,
+// failure is expected in the tests.
+//
+// We will use testing::Combine() to vary the first line: when we combine
+// AllCapabilities() with a single remainder assembly, we generate enough test
+// cases to try the assembly with every possible capability that could be
+// declared. However, Combine() only produces tuples -- it cannot produce, say,
+// a struct.  Therefore, this type must be a tuple.
+//
+// The first string is a capability name for the first line, the second the
```

Comment:
Wouldn't that be confusing, before we explained what "capability", "remainder", and "vector of must-pass" mean?